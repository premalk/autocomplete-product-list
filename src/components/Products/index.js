import { array, func } from 'prop-types';

// eslint-disable-next-line react/prop-types
const Products = ({ data, handleClick }) => {
  return (
    <>
      <table className="styled-table">
        <thead>
          <tr>
            <th>Title</th>
            <th>Gtin</th>
            <th>Gender</th>
            <th>Price</th>
            <th>Sale price</th>
            <th>Image link</th>
            <th>Additional images</th>
          </tr>
        </thead>
        <tbody>
          {data &&
            // eslint-disable-next-line react/prop-types
            data.map((obj, index) => {
              const { title, gtin, gender, sale_price, price, image_link, additional_image_link } =
                obj;

              return (
                <tr key={index} data-testid={title}>
                  <td>{title}</td>
                  <td>{gtin}</td>
                  <td>{gender}</td>
                  <td>{sale_price}</td>
                  <td>{price}</td>
                  <td>
                    <img src={image_link} />
                  </td>
                  <td>
                    <button onClick={() => handleClick(additional_image_link)}>
                      Additional images
                    </button>
                  </td>
                </tr>
              );
            })}
        </tbody>
      </table>
    </>
  );
};

Products.prototypes = {
  data: array,
  handleClick: func
};

export default Products;
