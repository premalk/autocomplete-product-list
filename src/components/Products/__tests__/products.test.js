import { render } from '@testing-library/react';
import Products from '../index';

const data = [
  {
    title: 'Weekday THURSDAY Jeans Slim Fit black',
    gtin: '2001007926858',
    gender: 'female',
    sale_price: '39.95 EUR',
    price: '39.95 EUR',
    image_link:
      'https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.4.jpg',
    additional_image_link:
      'https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@18.jpg'
  },
  {
    title: 'Weekday THURSDAY Jeans Slim Fit white',
    gtin: '2001007926858',
    gender: 'female',
    sale_price: '39.95 EUR',
    price: '39.95 EUR',
    image_link:
      'https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.4.jpg',
    additional_image_link:
      'https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@18.jpg'
  }
];

describe('Testing product list', () => {
  const products = render(<Products data={data} />);

  test('Snapshot Test of product', () => {
    expect(products).toMatchSnapshot();
  });
  it('products existance test', () => {
    const { getByTestId } = render(<Products data={data} />);

    expect(getByTestId('Weekday THURSDAY Jeans Slim Fit black')).toBeInTheDocument();
  });
});
