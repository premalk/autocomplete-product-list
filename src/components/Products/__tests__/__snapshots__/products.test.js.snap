// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing product list Snapshot Test of product 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="styled-table"
      >
        <thead>
          <tr>
            <th>
              Title
            </th>
            <th>
              Gtin
            </th>
            <th>
              Gender
            </th>
            <th>
              Price
            </th>
            <th>
              Sale price
            </th>
            <th>
              Image link
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="Weekday THURSDAY Jeans Slim Fit black"
          >
            <td>
              Weekday THURSDAY Jeans Slim Fit black
            </td>
            <td>
              2001007926858
            </td>
            <td>
              female
            </td>
            <td>
              39.95 EUR
            </td>
            <td>
              39.95 EUR
            </td>
            <td>
              <img
                src="https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.4.jpg"
              />
            </td>
          </tr>
          <tr
            data-testid="Weekday THURSDAY Jeans Slim Fit white"
          >
            <td>
              Weekday THURSDAY Jeans Slim Fit white
            </td>
            <td>
              2001007926858
            </td>
            <td>
              female
            </td>
            <td>
              39.95 EUR
            </td>
            <td>
              39.95 EUR
            </td>
            <td>
              <img
                src="https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.4.jpg"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="styled-table"
    >
      <thead>
        <tr>
          <th>
            Title
          </th>
          <th>
            Gtin
          </th>
          <th>
            Gender
          </th>
          <th>
            Price
          </th>
          <th>
            Sale price
          </th>
          <th>
            Image link
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          data-testid="Weekday THURSDAY Jeans Slim Fit black"
        >
          <td>
            Weekday THURSDAY Jeans Slim Fit black
          </td>
          <td>
            2001007926858
          </td>
          <td>
            female
          </td>
          <td>
            39.95 EUR
          </td>
          <td>
            39.95 EUR
          </td>
          <td>
            <img
              src="https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.4.jpg"
            />
          </td>
        </tr>
        <tr
          data-testid="Weekday THURSDAY Jeans Slim Fit white"
        >
          <td>
            Weekday THURSDAY Jeans Slim Fit white
          </td>
          <td>
            2001007926858
          </td>
          <td>
            female
          </td>
          <td>
            39.95 EUR
          </td>
          <td>
            39.95 EUR
          </td>
          <td>
            <img
              src="https://mosaic01.ztat.net/vgs/media/large/WE/B2/1N/00/HQ/11/WEB21N00H-Q11@12.4.jpg"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
